odt-opt-in/REPWRITERSPECS/BANNO.ODTOPTIN.V1.POW

```diff
@@ -1447,7 +1497,89 @@ PROCEDURE PRINTPROGRAMINFO
  NEWLINE
  PRINT "  "+"},"
  NEWLINE
-END
+END [PROCEDURE]
+
+PROCEDURE BNOPRINTRGDATA
+[* Print @RGUSER[CHR,NUM] data passed from the UX in proper JSON format
+**
+**  PASS    BNODEBUGMODE        Boolean: Print debug data, T/F
+*]
+ IF BNODEBUGMODE=BNODEBUGMODEON AND
+    BNOINCLUDERGDATAINDEBUG=TRUE THEN
+  DO
+   PRINT ","
+   NEWLINE
+   PRINT BNOINDENT(1)+Q+"rgData"+Q+": ["
+   NEWLINE
+   PRINT BNOINDENT(2)+Q+"CHR1:'"+SEGMENT(@RGUSERCHR1,1,120)+"'"+Q+","
+   NEWLINE
+   PRINT BNOINDENT(2)+Q+"CHR2:'"+SEGMENT(@RGUSERCHR2,1,120)+"'"+Q+","
+   NEWLINE
+   PRINT BNOINDENT(2)+Q+"CHR3:'"+SEGMENT(@RGUSERCHR3,1,120)+"'"+Q+","
+   NEWLINE
+   PRINT BNOINDENT(2)+Q+"CHR4:'"+SEGMENT(@RGUSERCHR4,1,120)+"'"+Q+","
+   NEWLINE
+   PRINT BNOINDENT(2)+Q+"CHR5:'"+SEGMENT(@RGUSERCHR5,1,120)+"'"+Q+","
+   NEWLINE
+   PRINT BNOINDENT(2)+Q+
+                     "NUM1:"+FORMAT("#########9 ",@RGUSERNUM1)+
+                     "NUM2:"+FORMAT("#########9 ",@RGUSERNUM2)+
+                     "NUM3:"+FORMAT("#########9 ",@RGUSERNUM3)+
+                     "NUM4:"+FORMAT("#########9 ",@RGUSERNUM4)+
+                     "NUM5:"+FORMAT("#########9",@RGUSERNUM5)+Q
+   NEWLINE
+   PRINT BNOINDENT(1)+"]"
+   NEWLINE
+  END
+END [PROCEDURE]
+
+PROCEDURE BNOPRINTDEBUG
+[* Print debug output in proper JSON format. Requires BNODEBUGMODE to be
+** on (or true).
+**
+**  PASS  BNODEBUGMODE        Boolean: Print debug data, T/F
+**  PASS  BNODEBUGMSGLIST(A)  Array of debug messages to print
+**  PASS  BNODEBUGCOUNT       Number of debug lines to print
+*]
+ IF BNODEBUGMODE=BNODEBUGMODEON THEN
+  DO
+   PRINT ","
+   NEWLINE
+   PRINT BNOINDENT(2)+Q+"debugData"+Q+": ["
+   NEWLINE
+   PRINT BNOINDENT(2)+Q+"Debug note: Debug lines truncated to 120 characters"+Q
+   IF BNODEBUGCOUNT>0 THEN
```

Comment:
Not sure you need this BNODEBUGCOUNT check since the below loop will just go from 1 to 0 (no iterations) if BNODEBUGCOUNT is 0.

Reply:
Removed